### PR TITLE
feat(amplication-client+server) added Public to EnumEntityPermissionT…

### PR DESCRIPTION
**Pull request by abrl91**
Issue Number: #2006 

The first part of this task:
Make sure the client is showing the option of public routes for every action and sending the correct values to the server.
The server has to recognize those values (in this case:  the new type - public). In order for it to happen, I added 'Public' to EnumEntityPermissionType enum and model and also to Prisma + migrate.

image

image

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-client/src/Permissions/EntityPermissionAction.tsx [hunk 1]
```diff
@@ -230,6 +231,10 @@ export const EntityPermissionAction = ({
                 <span className={`${CLASS_NAME}__action-summary`}>
                   {selectedRoleIds.size} roles selected
                 </span>
+              ) : permission.type === models.EnumEntityPermissionType.Public ? (
+                <span className={`${CLASS_NAME}__action-summary`}>
+                  Public endpoint selected
```
Comment (yuval-hazaz) [inline packages/amplication-client/src/Permissions/EntityPermissionAction.tsx:236]:
```suggestion
                  This action is public 
```

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]
Beside a small update in the display text, looks perfect!

File: packages/amplication-data-service-generator/src/server/create-grants.ts [hunk 2]
```diff
@@ -92,6 +93,18 @@ export function createGrants(entities: Entity[], roles: Role[]): Grant[] {
           }
           break;
         }
+        case EnumEntityPermissionType.Public: {
+          for (const role of roles) {
+            grants.push({
+              role: role.name,
+              resource: entity.name,
+              action: actionToACLAction[permission.action],
+              /** @todo */
```
Comment (tupe12334) [inline packages/amplication-data-service-generator/src/server/create-grants.ts:102]: What is this TODO?

Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/create-grants.ts:102]: This is a copy & paste from the AllRoles case for now until we know how to handle this case. When I only `break` on this case the, I am getting errors on react admin after login

**Comment by abrl91** [comment]
will open a new branch and a new pr to be more organized.

**Pull request closed without merging**